akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala

```diff
@@ -128,6 +133,21 @@ private[remote] abstract class InboundCompression[T >: Null](
         activeVersion, nextTable.version, incomingTableVersion)
       OptionVal.None
     }
+  }
+
+  def confirmAdvertisement(tableVersion: Int): Unit = {
```

Comment:
It's still using the first message, also.
The problem was that there might not be any such first message and then the whole compression process stalls.
- sender might not use any of the refs in the table, only using new refs
- adv msg might be lost